homeassistant/components/vodafone_station/device_tracker.py

```diff
@@ -0,0 +1,133 @@
+    )
+
+    update_router()
+
+
+@callback
+def add_entities(
+    coordinator: VodafoneStationRouter,
+    async_add_entities: AddEntitiesCallback,
+    tracked: set[str],
+) -> None:
+    """Add new tracker entities from the router."""
+    new_tracked = []
+
+    _LOGGER.debug("Adding device trackers entities")
+    for mac, device in coordinator.data["devices"].items():
+        if mac in tracked:
+            continue
+        _LOGGER.debug("New device tracker: %s", device.hostname)
+        new_tracked.append(VodafoneStationTracker(coordinator, device))
+        tracked.add(mac)
+
+    async_add_entities(new_tracked)
+
+
+class VodafoneStationTracker(CoordinatorEntity[VodafoneStationRouter], ScannerEntity):
+    """Representation of a Vodafone Station device."""
+
+    def __init__(
+        self, coordinator: VodafoneStationRouter, device: VodafoneStationDeviceInfo
+    ) -> None:
+        """Initialize a Vodafone Station device."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+        self._device_mac = device._mac
+        self._attr_unique_id = device._mac
+        self._attr_name = device._name or DEFAULT_DEVICE_NAME
+
+    @property
+    def _device_info(self) -> VodafoneStationDeviceInfo:
+        """Return fresh data for the device."""
+        return self.coordinator.data["devices"][self._device_mac]
+
+    @property
+    def is_connected(self) -> bool:
+        """Return true if the device is connected to the network."""
+        return self._device_info.is_connected
+
+    @property
+    def source_type(self) -> SourceType:
+        """Return the source type."""
+        return SourceType.ROUTER
+
+    @property
+    def hostname(self) -> str | None:
+        """Return the hostname of device."""
+        return self._attr_name
+
+    @property
+    def icon(self) -> str:
+        """Return device icon."""
+        return (
+            "mdi:lan-connect"
+            if self._device_info.is_connected
+            else "mdi:lan-disconnect"
+        )
+
+    @property
+    def ip_address(self) -> str | None:
+        """Return the primary ip address of the device."""
+        return self._device_info.ip_address
+
+    @property
+    def mac_address(self) -> str:
+        """Return the mac address of the device."""
+        return self._device_info.mac_address
+
+    @property
+    def extra_state_attributes(self) -> Mapping[str, Any] | None:
+        """Return additional attributes of the device."""
+        self._attr_extra_state_attributes = {}
+        self._attr_extra_state_attributes[
+            "connection_type"
+        ] = self._device_info.connection_type
+        if "Wifi" in self._device_info.connection_type:
+            self._attr_extra_state_attributes["wifi_band"] = self._device_info.wifi
+        self._attr_extra_state_attributes[
+            "last_time_reachable"
+        ] = self._device_info.last_activity
```

Comment:
https://developers.home-assistant.io/docs/core/entity?_highlight=entity#generic-properties

> Entities that generate a significant amount of state changes can quickly increase the size of the database when the extra_state_attributes also change frequently. Minimize the number of extra_state_attributes for these entities by removing non-critical attributes or creating additi

Reply:
Its still a problem even if other places are doing it, we shouldn't add more

Reply:
So we need to create a sensor for each device tracker ? I expect a lot of complains from users.
And it will be a breaking change from all other integrations.

Reply:
Humm, ok but i don't know how to calculate the `consider_home` then :-(

Reply:
Reworked based on your comment below